File: src/chipschallenge/blockreactions/HintFrom.java
package chipschallenge.blockreactions;

import chipschallenge.Block;
import chipschallenge.Inventory.Boots;

/**
 *
 * @author patrik
 */
public class HintFrom extends BlockReaction {

    private HintFrom() {
    }
    private static HintFrom mInstance = null;

    public static synchronized HintFrom getInstance() {
        if (mInstance == null) {
            mInstance = new HintFrom();
        }
        return mInstance;
    }

    public void react(Block moving, Block standing) {
        game().showHint();
    }

    public boolean canMove(Block moving, Block standing) {
        return true;
    }
}

File: ChipsChallenge.Shared/Blockreactions/HintFrom.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    public class HintFrom : NoSlipReaction
    {
        private static HintFrom instance;

        private HintFrom()
        {
        }

        public static HintFrom Instance
        {
            get
            {
                lock (typeof(HintFrom))
                {
                    return instance ?? (instance = new HintFrom());
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
            Game().IsChipOnHintField = false;
        }

        public override bool canMove(Block moving, Block standing)
        {
            return true;
        }
    }
}